server/services/userAdminService.js

```diff
@@ -108,11 +111,57 @@ module.exports = function createUserService(nomisClientBuilder, userClient) {
     }
   }
 
+  async function getIncompleteRoUsers(token) {
+    const incomplete = await userClient.getIncompleteRoUsers()
+    if (isEmpty(incomplete)) {
+      return []
+    }
+
+    const bookingIds = incomplete.map(u => u.bookingId)
+    const offenders = await getOffenderNomis(token, bookingIds)
+
+    return addMappingDetails(incomplete, offenders)
+  }
+
+  async function getOffenderNomis(token, bookingIds) {
+    try {
+      const nomisClient = nomisClientBuilder(token)
+      const offenders = await nomisClient.getOffenderSentencesByBookingId(bookingIds, false)
+
+      if (isEmpty(offenders)) {
+        return null
+      }
+
```

Comment:
make a map of bookingId to offender number so that we can add offender number to the "incomplete user" object